controllers/auth/signup.js

```diff
@@ -0,0 +1,50 @@
+const jwt = require('jsonwebtoken');
+const randtoken = require('rand-token');
+const bcrypt = require('bcrypt');
+
+const { Users, Tokens } = require('../../db/models');
+
+const key = process.env.secretKey;
+
+async function signup(req, res) {
+  const {
+    body: {
+      firstName,
+      lastName,
+      avatar,
+      email,
+      password,
+    },
+  } = req;
+  if (email.trim() === '' || password.trim() === '' || firstName.trim() === '' || lastName.trim() === '') {
+    return res.status(400).send({ msg: 'Missed data.' });
```

Comment:
You need to check that email is really email - not a random string - you can use regexp for that